### ISSUE TITLE
x86 integrated assembler: .altmacro string replacement not performed in .irp macro use

### ISSUE THREAD

**Issue by emaste** [llvm/llvm-project#38347]
|  |  |
| --- | --- |
| Bugzilla Link | [38999](https://llvm.org/bz38999) |
| Version | 5.0 |
| OS | FreeBSD |

## Extended Description 
Found while trying to build the FreeBSD base system entirely with the integrated assembler. 

One of .altmacro's effects is:

Expression results as strings
    You can write `%expr' to evaluate the expression expr and use the result as a string. 

Consider the following snippet:
```asm
.altmacro

.macro m dst,src
  addq %\src,%\dst
.endm

.irp r,r15
  m r,r8
.endr
```

GNU as and gcc assemble this, while Clang (6.0) IAS produces:
```
<instantiation>:1:10: error: invalid register name
addq %r8,%r
         ^~
<instantiation>:1:1: note: while in macro instantiation
m r,r8
^
macro.s:7:1: note: while in macro instantiation
.irp r,r15
^
```

**Comment by MaskRay** [comment]
`%` expansion in the altmacro mode is performed for macro arguments instead of the expanded body.
LLVMMCParser incorrectly expands `%` for the expanded body.

**Comment by MaskRay** [comment]
Fixed by f8b1ca4992a22b4b65282c09dd6f07a1a2839070
